src/cephadm/cephadm

```diff
@@ -4466,7 +4467,11 @@ def prepare_mon_addresses(
     logger.debug('Base mon IP is %s, final addrv is %s' % (base_ip, addr_arg))
 
     mon_network = None
-    if not ctx.skip_mon_network:
+    cp = read_config(ctx.config)
+    if cp.has_option('global', 'public_network'):
+        mon_network = cp.get('global', 'public_network')
+
+    if mon_network is None and not ctx.skip_mon_network:
```

Comment:
One question. if we pass a config option for the public_network that does not work at all for the mon we deploy during bootstrap, is the cluster still operable? As in, can you still shell in and run commands afterwards even if the only existent monitor isn't actually in the public network?

Reply:
I simulated this situation just to see what happens. Basically I provided a conf file where no public network works. In this case cephadm is able to create a mon (because we have to provide a correct ip in the param `--mon-ip`) but afterwords it's not able to create/run a valid mgr. This happens because mgr look for a valid IP in the provided public_network list and since it doesn't find any it end up failing with a message similar to the following:

`unable to find any IP address in networks '<the provided public list>' interfaces ''
`
So in a real scenario user won't be able to continue with the deployment as the provided networks are not valid. **We need at least one valid network in the public list to create and run the mgr** so cephadm can continue the bootstrap.

Reply:
Do you think maybe if we have both --mon-ip/--mon-addrv and public_network specified in the config we can check that the address(es) from the flag matches with the public_network in the config? It wouldn't make sense to specify an ip for your mon that isn't in the public network and, as you just posted, the bootstrap ends up failing anyway if that's the case.

Reply:
seems like we should validate the `mon_ip` is within the range of at least one of the public CIDR networks 

likewise, maybe we should also ensure that each of the public CIDR networks are actually configured locally?

Reply:
I agree and I'm working on these changes but as part of this work I'm performing some refactoring to improve the readability of the `def prepare_mon_addresses`  code. I don't want to include these new changes in this PR. I'd suggest to merge these changes and submit the refactoring (+ new checks) as part of another PR.